src/os/unix/local_socket.rs

```diff
@@ -125,7 +125,7 @@ impl FromRawFd for LocalSocketStream {
 }
 
 #[inline]
-fn local_socket_name_to_ud_socket_path(name: LocalSocketName<'_>) -> io::Result<UdSocketPath<'_>> {
+fn local_socket_name_to_ud_socket_path(name: LocalSocketName) -> io::Result<UdSocketPath> {
```

Comment:
```suggestion
fn local_socket_name_to_ud_socket_path(name: LocalSocketName<'_>) -> io::Result<UdSocketPath<'_>> {
```
I wanted to do `#![deny(rust_2018_idioms)]` in `lib.rs` actually, forgot to do that before. I'll add it myself directly, after merging.

Reply:
Alright, out of curiosity, why do you want deny(rust_2018_idioms)?

Reply:
`rust_2018_idioms` is the lint which is triggered when you don't follow Rust 2018's new guidelines; hence, `#![deny(rust_2018_idioms)]` would raise them to errors by default. In this case, the `<'_>` is required by the guidelines because it makes it much more obvious that the type is borrowing something rather than adhering to `'static`. This is mostly a style preference, though it makes stuff much clearer sometimes.